earthpy/spatial.py

```diff
@@ -888,3 +924,119 @@ def draw_legend(im_ax, titles=None, cmap=None, classes=None, bbox=(1.05, 1)):
         borderaxespad=0.0,
         prop={"size": 13},
     )
+
+
+# @deprecate
+def stack_raster_tifs(band_paths, out_path, arr_out=True):
+    """Take a list of raster paths and turn into an output raster stack
+    numpy array. Note that this function depends upon the stack() function.
+
+    TODO: Instead of returning a file path when arr_out=False, consider
+        returning None since the out_path is already
+        an input given by the user. This will make the output type consistent.
+
+    Parameters
+    ----------
+    band_paths : list of file paths
+        A list with paths to the bands you wish to stack. Bands
+        will be stacked in the order given in this list.
+    out_path : string
+        A path with a file name for the output stacked raster
+         tif file.
+    arr_out : boolean
+        A boolean argument to designate what is returned in the stacked
+        raster tif output.
+
+    Returns
+    -------
+    If arr_out keyword is True:
+        tuple: The first value representing the result of src.read() of the
+        stacked array and the second value
+        representing the result of src.profile of the stacked array.
+    If arr_out keyword is False:
+        str : A path with a file name for the output stacked raster tif file.
+
+    
+    """
+
+    # Throw warning and exit
+    raise Warning("stack_raster_tifs is deprecated. Use stack(). Exiting...")
+    sys.exit()
+
+    # Set default import to read
+    kwds = {"mode": "r"}
+
+    out_dir = os.path.dirname(out_path)
+    writing_to_cwd = out_dir == ""
+    if not os.path.exists(out_dir) and not writing_to_cwd:
+        raise ValueError(
+            "The output directory path that you provided does not exist"
+        )
+
+    if len(band_paths) < 2:
+        raise ValueError(
+            """The list of file paths is empty. You need at least
+                            2 files to create a stack."""
+        )
+    with contextlib.ExitStack() as context:
+        sources = [
+            context.enter_context(rio.open(path, **kwds))
+            for path in band_paths
+        ]
+
+        # TODO: Check that the CRS and TRANSFORM are the same
+        dest_kwargs = sources[0].meta
+        dest_count = sum(src.count for src in sources)
+        dest_kwargs["count"] = dest_count
+
+        if arr_out:
+            # Write stacked gtif file
+            with rio.open(out_path, "w", **dest_kwargs) as dest:
+                _stack(sources, dest)
+            # Read and return array
+            with rio.open(out_path, "r") as src:
+                return src.read(), src.profile
+        else:
+            # Write stacked gtif file
+            with rio.open(out_path, "w", **dest_kwargs) as dest:
+                return _stack(sources, dest)
+
+
+# Function to be submitted to rasterio
```

Comment:
@joemcglinchy let's remote these comments. they are dated at this point. and you have added unit tests!!

Reply:
got it. done.